lib/elements/multiselect.js

```diff
@@ -136,19 +136,35 @@ class MultiselectPrompt extends Prompt {
     }
   }
 
+  toggleAll() {
+    if (this.maxChoices !== undefined) {
+      return this.bell();
+    }
+
+    this.value.forEach(v => v.selected = !v.selected);
```

Comment:
It will always take the opposite state here which might be problematic. If one item is selected, the selected one will be unselected and the rest selected.

Instead of just negate `v.selected` you could do something like this (not tested, just giving an idea here)

```js
const newSelectedState = !this.value[0].selected;
this.value.forEach(v => v.selected = newSelectedState);
```

That way, it always takes the negated state of the first element.

Reply:
The options we have here are:
1. Toggling all (currently implemented by the PR)
2. Set all to the inverse of the first (Your suggestion)
3. Set all to the inverse of the currently "hovered" (Similar to your suggestion)
4. Hold an additional boolean state that cycles between all on and all off regardless of what individual toggles happen in between (e.g. used in yarn upgrade-interactive)

I don't have any strong feelings here, other than 3 IMO being a bit smarter (and not harder to implement) than 2, and 4 introducing extra complexity into the multiselect that I'm not sure is worth it.

Feel free to pick one and I'll follow up when I have time ;)

Reply:
Good point and not a bad idea actually. I think however that it doesn't really matter if it's one of 2, 3 or 4 as long as all items are set to the same state 😊

I would say go ahead with 3 then, if it is easy to do.